### DISCUSSION TITLE
session reset

### DISCUSSION THREAD

**Discussion by supun19** [supabase/supabase#25744] [Questions]
in my application i am checking user has twitter account or not for that i am using twitter oauth2 api for verify it.

so user login to my application using email and password using supabase auth.
after login using i am using twitter sso for verify user. when doing that twitter reidrect url with my application "http://localhost:3000/withsocial?type=twitter&state=state&code=". then supbase hit request https://projectid.supabase.co/auth/v1/token?grant_type=pkce and it return 404 and session reset.
i am confusing how auth session being reset

**Comment by guzfernandez** [comment]
I know this is a late reply, but maybe it can be useful for future readers. 
I had the same problem, and what I found out is that if you pass `code` query parameter in the url, Supabase assumes it’s a PKCE/OAuth flow and tries to exchange it for a session token, which obviously fails. 
What you can do is use a different name for that param, for example `auth_code` or something like that. You could also remove the `code` param from the URL before Supabase tries to check it with its backend. 

UPDATE: I ended up redirecting the user to an /api endpoint instead of a page, since Supabase doesn't run its auth logic server side. Then redirect the user back to a page, but changing `code` to `auth_code`.

Let me know if you find this useful! :)
